### PR TITLE
Rework n_jobs parameter

### PR DESCRIPTION
**Pull request by weixuanfu2016**
This PR is for solving issues #436 and #490 related to `joblib` by:

1. `joblib` module is not used in TPOT when n_jobs is equal to 1

2. Rework `_wrapped_cross_val_score` in TPOT based on [the source codes of `sklearn.model_selection.cross_val_score`](https://github.com/scikit-learn/scikit-learn/blob/14031f6/sklearn/model_selection/_validation.py#L36) and remove the `joblib` module in `cross_val_score` to avoid nested usage of `joblib` in TPOT

### CODE REVIEW

File: tpot/base.py [hunk 1]
```diff
@@ -890,28 +890,31 @@ def _evaluate_individuals(self, individuals, features, target, sample_weight=Non
...
-                    max_eval_time_mins=self.max_eval_time_mins,
-                    groups=groups
-                )
-                jobs.append(job)
-            parallel = Parallel(n_jobs=self.n_jobs, verbose=0, pre_dispatch='2*n_jobs')
-            tmp_result_score = parallel(jobs)
-
+            if self.n_jobs == 1:
+                tmp_result_scores = [partial_wrapped_cross_val_score(sklearn_pipeline=sklearn_pipeline)
+                for sklearn_pipeline in sklearn_pipeline_list[chunk_idx:chunk_idx + self.n_jobs * 4]]
```
Comment (rhiever) [inline tpot/base.py]: This line should be indented more to make it clear that it's a continuation of the previous line.

Comment (rhiever) [inline tpot/base.py]: Also, there should be no chunking in the case where `n_jobs=1`, right?

File: tpot/base.py [hunk 2]
```diff
@@ -890,37 +890,37 @@ def _evaluate_individuals(self, individuals, features, target, sample_weight=Non
...
-                                         'Continuing to the next pipeline.'.format(self._pbar.n))
-                    resulting_score_list.append(-float('inf'))
+        if self.n_jobs == 1:
+            for sklearn_pipeline in sklearn_pipeline_list:
+                val = partial_wrapped_cross_val_score(sklearn_pipeline=sklearn_pipeline)
+                resulting_score_list = self._update_pbar(val, resulting_score_list)
+        else:
+            # chunk size for pbar update
+            for chunk_idx in range(0, len(sklearn_pipeline_list), self.n_jobs * 4):
+                if self.n_jobs == 1:
```
Comment (rhiever) [inline tpot/base.py]: This check doesn't need to be here.

File: tpot/base.py [hunk 3]
```diff
@@ -992,6 +992,19 @@ def _operator_count(self, individual):
                 operator_count += 1
         return operator_count
 
+    # update self._pbar during pipeline evaluration
```
Comment (rhiever) [inline tpot/base.py]: Please make this comment a docstring for the `_update_pbar` function.

Comment (weixuanfu2016) [inline tpot/base.py]: Okay, will do

**Pull request merged**